multy_transaction/src/transaction.cpp

```diff
@@ -0,0 +1,211 @@
+#include "multy_transaction/transaction.h"
+
+#include "multy_transaction/internal/transaction.h"
+#include "multy_transaction/internal/u_ptr.h"
+
+#include "multy_core/internal/utility.h"
+#include "multy_core/internal/account.h"
+
+namespace
+{
+using namespace wallet_core::internal;
+using namespace multy_transaction::internal;
+} // namespace
+
+Error* make_transaction(const Account* account, Transaction** new_transaction)
+{
+    ARG_CHECK(account);
+    ARG_CHECK(new_transaction);
+
+    try
+    {
+//        switch (account->get_currency())
+//        {
+//            case CURRENCY_BITCOIN:
+//                *new_transaction =
+//        }
+    }
+    CATCH_EXCEPTION_RETURN_ERROR();
+
+    OUT_CHECK(*new_transaction);
+
+    return nullptr;
+}
+
+Error* transaction_has_trait(
+        const Transaction* transaction,
+        TransactionTrait trait,
+        bool* out_has_capability)
+{
+    ARG_CHECK(transaction);
+    ARG_CHECK(
+            trait == TRANSACTION_REQUIRES_EXPLICIT_SOURCE
+            || trait == TRANSACTION_SUPPORTS_MULTIPLE_SOURCES
+            || trait == TRANSACTION_SUPPORTS_MULTIPLE_DESTINATIONS
+            || trait == TRANSACTION_SUPPORTS_CHANGE
+            || trait == TRANSACTION_SUPPORTS_FEE);
+
+    try
+    {
+        *out_has_capability = transaction->get_traits() & (1 << trait);
+    }
+    CATCH_EXCEPTION_RETURN_ERROR();
+
+    return nullptr;
+}
+
+Error* transaction_get_currency(
+        const Transaction* transaction, Currency* out_currency)
+{
+    ARG_CHECK(transaction);
+
```

Comment:
Add ARG_CHECK(out_currency)

Reply:
Not required here, since we don't allocate memory nor setting a pointer to pointer, just writing to given object.